src/viam/utils.py

```diff
@@ -77,23 +77,16 @@ def value_to_primitive(value: Value) -> Any:
 
 def resource_names_for_component(component: ComponentBase) -> List[ResourceName]:
```

Comment:
somewhat unrelated to this PR but I'm pretty confused about what this function does - don't we only have one resource name per resource? and we can construct it with `component.get_resource_name(component.name)`?

Reply:
Components can technically have multiple resource names -- say I have a custom component that implements both Servo and Gripper, it could theoretically have 2 resource names (I should be able to get it as either `rdk:component:servo:custom` or `rdk:component:gripper:custom`

Reply:
ah yes I remember this now :)